tests/workers/test_process_worker.py

```diff
@@ -348,761 +369,151 @@ async def test_flow_run_vars_and_deployment_vars_get_merged(
+            configuration=configuration,
         )
 
         assert isinstance(result, ProcessWorkerResult)
         assert result.status_code == 0
-        assert mock.call_args.args == (correct_default,)
+        mock_runner_execute_flow_run.assert_awaited_once_with(
+            flow_run_id=flow_run.id,
+            command=configuration.command,
+            cwd=configuration.working_dir,
+            env=configuration.env,
+            stream_output=configuration.stream_output,
+            task_status=None,
+        )
 
 
 async def test_process_worker_command_override(
-    flow_run, patch_run_process, process_work_pool, monkeypatch
+    deployment_with_overrides: Deployment,
+    flow_run_with_deployment_overrides: FlowRun,
+    mock_runner_execute_flow_run: MagicMock,
+    process_work_pool: WorkPool,
+    monkeypatch: pytest.MonkeyPatch,
 ):
-    mock: AsyncMock = patch_run_process()
-    override_command = "echo hello world"
-    override = {"command": override_command}
-    patch_client(monkeypatch, overrides=override)
-
+    override_command = deployment_with_overrides.job_variables["command"]
     async with ProcessWorker(work_pool_name=process_work_pool.name) as worker:
-        worker._work_pool = process_work_pool
+        configuration = await worker._get_configuration(
+            flow_run_with_deployment_overrides
+        )
         result = await worker.run(
-            flow_run=flow_run,
-            configuration=await worker._get_configuration(flow_run),
+            flow_run=flow_run_with_deployment_overrides,
+            configuration=configuration,
         )
 
         assert isinstance(result, ProcessWorkerResult)
         assert result.status_code == 0
-        assert mock.call_args.args == (override_command.split(" "),)
+        mock_runner_execute_flow_run.assert_awaited_once_with(
+            flow_run_id=flow_run_with_deployment_overrides.id,
+            command=override_command,
+            cwd=configuration.working_dir,
+            env=configuration.env,
+            stream_output=configuration.stream_output,
+            task_status=None,
+        )
 
 
-async def test_task_status_receives_infrastructure_pid(
-    process_work_pool, patch_run_process, monkeypatch, flow_run
+async def test_task_status_receives_pid(
+    process_work_pool: WorkPool,
+    flow_run: FlowRun,
 ):
-    patch_client(monkeypatch)
     fake_status = MagicMock(spec=anyio.abc.TaskStatus)
     async with ProcessWorker(work_pool_name=process_work_pool.name) as worker:
-        worker._work_pool = process_work_pool
         result = await worker.run(
             flow_run=flow_run,
             configuration=await worker._get_configuration(flow_run),
             task_status=fake_status,
         )
 
-        hostname = socket.gethostname()
-        fake_status.started.assert_called_once_with(f"{hostname}:{result.identifier}")
-
-
-async def test_process_kill_mismatching_hostname(monkeypatch, process_work_pool):
```

Comment:
I removed these tests since the process worker doesn't handle cancellation any longer.